### ISSUE TITLE
Script doesn't fetch updatet file

### ISSUE THREAD

**Issue by RakeshChotkan** [derekantrican/GAS-ICS-Sync#404]
### The problem

The script doesn't fetch an updated version of the ics url.
I have an own domain where I upload the ICS file. First I uploaded a file with 7 events. The first time when the script runs, it fetches the 7 events and add it succesfully the events to the specified calendar.

When I upload an ICS file with 13 events, which are in the future, the script still fetches 7 events instead of the 13 events. I am sure the filename of the new upload file is the same because when I open the url by myself in the browser it opens the updated file.

Below are the settings of my configuration.

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/47205952/4be6c377-3efa-4f85-b1ba-3404e5a32b71)

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/47205952/fad865f8-4828-4e90-bc62-da07cc49afd3)

I am 100% sure I have uploaded the right file and checked everything more than 10 times, but it looks like it holding the first file in chache or something because it doesn't fetch the updated file.

### Version of GAS-ICS-Sync

5.8

**Comment by jonas0b1011001** [comment]
The script itself does not store any data about the calendars.
There are also no indications that [UrlFetchApp](https://developers.google.com/apps-script/reference/url-fetch/url-fetch-app), used to retrieve the source files, uses any kind of caching.

What happens if you set up a new copy of the script any sync the same url?

**Comment by RakeshChotkan** [comment]
Thank you [jonas0b1011001](https://github.com/jonas0b1011001) for your quick response. I am also a developer and I was 100% sure I wasn't doing anything wrong and it isn't also not a flaw of the script.

After further investigation I came to the conclusion that Google itself was keeping the file in cache. Initially I was storing the updated file on Wordpress (in a subdirectory), but the website itself allows to keep a cache I think. After your response I also tried an another instance but with the same issue. Then I stored the file on another server and when the is updated now, the changes are captured immediately.

So it took me quite a lot of hours to figure it out, but for everyone else. If your updated file doesn't change the events/calendar, just change the location of the ics file because there is a change Google kept it cached on their own servers.
